### PR TITLE
testing account app serializer

### PR DESCRIPTION
**Pull request by Msdot001**
# Description

I tested accounts app serializers using a simple pytest unit test. 

I made a slight modification to the user serialization and wrote a method to automatically count the number of accounts a user has.

### CODE REVIEW

**Comment by Msdot001** [comment]
31/01/2023
1. Continue writing code to test the accounts app serializer and views. 

2. Made modifications to UserAccountRegistrationSerializer, but it still serves the same purpose. 

-- New serializer--

class UserAccountRegistrationSerializer(serializers.ModelSerializer):
    confirm_password = serializers.CharField(style={"input_type": "password"}, write_only=True)
    user = UserDetailsSerializer()

    def create(self, validated_data):
        user = dict(validated_data.pop('user'))
        password = user.pop('password')
        confirm_password = validated_data.pop('confirm_password')

        if password == confirm_password:
            with transaction.atomic():
                user = User.objects.create_user(**user, password=password)

                account = Account.objects.create(user=user, **validated_data)

                return account
            
    class Meta:
        model = Account
        fields = ['id', 'user', 'confirm_password', 'account_name', 'display_picture', 'bio']
 

-- Old serializer--

class UserAccountRegistrationSerializer(serializers.ModelSerializer):
    confirm_password = serializers.CharField(style={"input_type": "password"}, write_only=True)
    user = UserDetailsSerializer()

    def create(self, validated_data):
        user = dict(self.validated_data['user'])
        password = user['password']
        confirm_password = self.validated_data['confirm_password']

        if password == confirm_password:
            with transaction.atomic():
                user = User.objects.create_user(username=user['username'], first_name=user['first_name'], last_name=user['last_name'], email=user['email'], phone_number=user['phone_number'], password=password)

                account = Account.objects.create(user=user, account_name=self.validated_data['account_name'], bio=self.validated_data['bio'])

                return account
                
    class Meta:
        model = Account
        fields = ['id', 'user', 'confirm_password', 'account_name', 'display_picture', 'bio']

3. Encounter problem test RegisterAccountViewSet but other views were tested.

**Comment by hellen-22** [comment]
> 

Why do we need to write a method to count number of accounts, yet the number of accounts attribute/field was annotated in the UserViewset?

**Comment by Msdot001** [comment]
@hellen-22: I will look into it.

File: accounts/serializers.py [hunk 1]
```diff
@@ -12,13 +12,19 @@ class Meta:
 
 class UserSerializer(serializers.ModelSerializer):
     password = serializers.CharField(write_only=True)
-    number_of_accounts = serializers.IntegerField(read_only=True) #Added field to display the number of accounts that a user has.
+    #number_of_accounts = serializers.IntegerField(read_only=True) #Added field to display the number of accounts that a user has.
+    number_of_accounts = serializers.SerializerMethodField()
```
Comment (JimmyTron) [inline accounts/serializers.py:16]: no foreseable error

**Review by JimmyTron — COMMENTED** [review]
Looks good to go 👍👍

**Review by JimmyTron — APPROVED** [review]
Could be merged

**Pull request merged**